cpp/include/cuspatial/experimental/point_distance.cuh

```diff
@@ -18,40 +18,28 @@
 
 #include <rmm/cuda_stream_view.hpp>
 
+#include <cuspatial/experimental/array_view/multipoint_array.cuh>
+
 namespace cuspatial {
 
 /**
  * @ingroup distance
  * @copybrief cuspatial::pairwise_point_distance
  *
- * @tparam Cart2dItA iterator type for point array of the first point of each pair. Must meet
- * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
- * @tparam Cart2dItB iterator type for point array of the second point of each pair. Must meet
- * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
- * @tparam OutputIt iterator type for output array. Must meet the requirements of
- * [LegacyRandomAccessIterator][LinkLRAI], be mutable and be device-accessible.
- *
- * @param points1_first beginning of range of the first point of each pair
- * @param points1_last end of range of the first point of each pair
- * @param points2_first beginning of range of the second point of each pair
- * @param distances_first beginning iterator to output
- * @param stream The CUDA stream to use for device memory operations and kernel launches
- * @return Output iterator to one past the last element in the output range
+ * @tparam MultiPointArrayViewA An instance of template type `array_view::multipoint_array`
+ * @tparam MultiPointArrayViewB An instance of template type `array_view::multipoint_array`
  *
- * @pre all input iterators for coordinates must have a `value_type` of `cuspatial::vec_2d`.
- * @pre all scalar types must be floating point types, and must be the same type for all input
- * iterators and output iterators.
- *
- * [LinkLRAI]: https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator
- * "LegacyRandomAccessIterator"
+ * @param multipoints1 Array view object constructed from the iterators of the first multipoint in
+ * the pair.
+ * @param multipoints2 Array view object constructed from the iterators of the second multipoint in
+ * the pair.
```

Comment:
I don't think the docs need to dictate how or from what the view/span was constructed.

```suggestion
 * @param multipoints1 Span of first multipoints in each distance pair.
 * @param multipoints2 Span of second multipoints in each distance pair.
```

Reply:
Agreed except shouldn't call it span. Maybe reference object?